### PR TITLE
Refactoring of doors and levels game

### PR DESCRIPTION
**Pull request by spore9**
#54

### CODE REVIEW

File: HW2/Kirill_Uryvaev/DoorsAndLevelsGame/GameSettings.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        public int MaxDoorNumber;
+
+        public GameSettings(List<string> settings)
+        {
+            if (settings.Count!= settingsNumber)
+            {
+                throw new Exception("Invalid settings number");
+            }
+
+            switch (settings[0])
```
Comment (chesdenis) [inline HW2/Kirill_Uryvaev/DoorsAndLevelsGame/GameSettings.cs]: what if we want to add new languages without recompiling our game?

Comment (spore9) [reply HW2/Kirill_Uryvaev/DoorsAndLevelsGame/GameSettings.cs]: Oh, thanks, I didn't even think about it

File: HW2/Kirill_Uryvaev/DoorsAndLevelsGame/GameManager.cs [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+            _settings = settingsProvider.GetSettings();
+            phraseProvider.SetLanguage(_settings.LanguageFile);
+            isRestarting = false;          
+        }
+
+        public void Run()
+        {
+            createRandomDoors();
+            inputOutputProvider.Write(phraseProvider.GetPhrase("Rules"));
```
Comment (vsatkov) [inline HW2/Kirill_Uryvaev/DoorsAndLevelsGame/GameManager.cs]: You already have not "zero" in _settings.ExitDoorNumber. You should change rules or another part of code.

File: HW2/Kirill_Uryvaev/DoorsAndLevelsGame/Resources/Localization/English.json [hunk 3]
```diff
@@ -1,5 +1,5 @@
 ﻿{
-  "Rules": "########################################### \n The rules: \n To exit the game, enter 'e' character. \n To go on the next level enter door number. \n To go on the previous level enter zero. \n If you already at zero level, \n Going on previous level will regenerate doors \n ###########################################",
+  "Rules": "########################################### \n The rules: \n To exit the game, enter \"@ExitCode\". \n To go on the next level enter door number. \n To go on the previous level enter @ExitDoor. \n If you already at zero level, \n Going on previous level will regenerate doors \n ###########################################",
```
Comment (vsatkov) [inline HW2/Kirill_Uryvaev/DoorsAndLevelsGame/Resources/Localization/English.json]: Perfect decision. Thanks

**Pull request merged**